### ISSUE TITLE
[feature] 2 Factor Authentication (Fingerprint + pin)

### ISSUE THREAD

**Issue by iuginP** [PhilippC/keepass2android#399]
It would be great a 2FA after the fingerprint, for enhanced security: each one should use long complex password to protect the database, but when using the smartphone this could be problematic. So the fingerprint is a great idea, but anybody could force use to use the fingerprint (or use it inappropriately while we are, for example, unconscious or drunk).
A great solution could be, for example, to require a **pin after the fingerprint** (less complex then the password, but ensuring robust security).

**Comment by 1688pc** [comment]
吓得我赶紧把支付宝指纹支付关了

**Comment by vertigo220** [comment]
I don't think it would help if you're being forced to use your fingerprint, since in such a case you'd also be forced to provide the PIN (a panic PIN would be helpful in such a case, with options when entered to delete the database and to send a text with your location and a user-specified message to user-selected numbers; paranoid, sure, but why not). However, it would be extremely helpful to make it easier and quicker to unlock the database without making it as easy as just a fingerprint. My main concern with using the fingerprint is that the password is stored in the Android KeyStore, which I don't really trust to keep it secure. I don't know if a PIN would just be stored there as well, thereby adding no additional security in that sense, or if it could be stored elsewhere, which _would_ then make the use of fingerprint unlock more secure.

**Comment by iuginP** [comment]
@PhilippC I think this and #644 are the same... maybe should be merged?

**Comment by Handwrit** [comment]
Even better, how about 3FA?

Password + YubiKey OTP + Fingerprint

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
> Even better, how about 3FA?
> 
> Password + YubiKey OTP + Fingerprint

Don't forget the keyfile 🤭
